src/main/java/echoserver/EchoServer.java

```diff
@@ -2,25 +2,41 @@
 
 import java.io.IOException;
 import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
 
 public class EchoServer {
     private final ServerSocket serverSocket;
+    private ExecutorService executor;
+    private ArrayList<Socket> clientSockets = new ArrayList<>();
 
-    EchoServer(ServerSocket serverSocket) {
+
+    EchoServer(ServerSocket serverSocket, ExecutorService executor) {
         this.serverSocket = serverSocket;
+        this.executor = executor;
     }
 
-    public void start() {
-        System.out.println("Server started");
-        listenForClients();
+    void start() {
+        while (true) listenForClients();
     }
 
-    public void listenForClients() {
+    void listenForClients() {
         try {
+            System.out.println("Listening for clients...");
```

Comment:
avoid direct `System.out` calls.

Reply:
Thanks. In a later commit,  I created a new class, `ConsoleWriter`, with a `println` method to replace this